### PR TITLE
Update swc in next-swc

### PR DESCRIPTION
**Pull request by padmaia**
Updated swc and started using new `process_js_with_custom_pass` function and error handling system.

### CODE REVIEW

File: packages/next/build/swc/src/minify.rs [hunk 1]
```diff
@@ -29,26 +29,59 @@ use crate::{
...
+
+impl MinifyTarget {
+  fn to_file(&self, cm: Lrc<SourceMap>) -> Lrc<SourceFile> {
+    match self {
+      MinifyTarget::Single(code) => cm.new_source_file(FileName::Anon, code.clone()),
+      MinifyTarget::Map(codes) => {
+        assert_eq!(
+          codes.len(),
+          1,
+          "swc.minify does not support concatting multiple files yet"
```
Comment (styfle) [inline packages/next/build/swc/src/minify.rs:62]:
```suggestion
          "swc.minify does not support concatenating multiple files yet"
```

🐱

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| buildDuration | 14.2s | 14.2s | ⚠️ +24ms |
| buildDurationCached | 3.6s | 3.6s | -22ms |
| nodeModulesSize | 47 MB | 47 MB | ⚠️ +4 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.748 | 2.741 | -0.01 |
| / avg req/sec | 909.74 | 912.06 | +2.32 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.482 | 1.478 | 0 |
| /error-in-render avg req/sec | 1686.55 | 1691.12 | +4.57 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| 745.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 26.6 kB | 26.6 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.4 kB | 70.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| _app-HASH.js gzip | 979 B | 979 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.14 kB | 4.14 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 318 B | 318 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| _buildManifest.js gzip | 492 B | 492 B | ✓ |
| Overall change | 492 B | 492 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| index.html gzip | 541 B | 541 B | ✓ |
| link.html gzip | 553 B | 553 B | ✓ |
| withRouter.html gzip | 534 B | 534 B | ✓ |
| Overall change | 1.63 kB | 1.63 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| buildDuration | 10.9s | 10.9s | -17ms |
| buildDurationCached | 4.6s | 4.6s | -21ms |
| nodeModulesSize | 47 MB | 47 MB | ⚠️ +4 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.675 | 2.677 | 0 |
| / avg req/sec | 934.52 | 933.93 | ⚠️ -0.59 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.391 | 1.416 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1797.78 | 1765.37 | ⚠️ -32.41 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| 16.HASH.js gzip | 186 B | 186 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 14.1 kB | 14.1 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| webpack-HASH.js gzip | 1.19 kB | 1.19 kB | ✓ |
| Overall change | 71.2 kB | 71.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| _app-HASH.js gzip | 964 B | 964 B | ✓ |
| _error-HASH.js gzip | 3.8 kB | 3.8 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.87 kB | 2.87 kB | ✓ |
| head-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks-HASH.js gzip | 924 B | 924 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.03 kB | 3.03 kB | ✓ |
| withRouter-HASH.js gzip | 295 B | 295 B | ✓ |
| 30809af5c834..565.css gzip | 125 B | 125 B | ✓ |
| Overall change | 18.1 kB | 18.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | padmaia/next.js chore/update-swc | Change |
| - | - | - | - |
| index.html gzip | 585 B | 585 B | ✓ |
| link.html gzip | 596 B | 596 B | ✓ |
| withRouter.html gzip | 579 B | 579 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

</details>
Commit: df3c964d9022d842d550d0a4cc7e38e2c09c1885

**Comment by padmaia** [comment]
The rust code is for the most part copy pasted from swc, except I removed some code we don't need and am using `process_js_with_custom_pass` instead of `process_js` if that makes anyone more comfortable reviewing.

**Review by styfle — DISMISSED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: df3c964d9022d842d550d0a4cc7e38e2c09c1885

**test/integration/create-next-app/index.test.js** 
- create next app > should allow example with GitHub URL
- create next app > should allow example with GitHub URL and example-path
- create next app > should use --example-path over the file path in the GitHub URL
- create next app > should use npm as the package manager on supplying --use-npm with example

<details>
<summary>Expand output</summary>

  ● create next app › should allow example with GitHub URL

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-app --example https://github.com/vercel/next-learn-starter/tree/master/navigate-between-pages-starter

      136 |       ).toBeTruthy()
      137 |
    > 138 |       // Assert for dependencies specific to the typescript template
          |                         ^
      139 |       const pkgJSON = require(pkgJSONPath)
      140 |       expect(Object.keys(pkgJSON.dependencies)).toEqual([
      141 |         'next',

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:138:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:136:9)

  ● create next app › should allow example with GitHub URL and example-path

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-example-path --example https://github.com/vercel/next-learn-starter/tree/master --example-path navigate-between-pages-starter

      153 |   })
      154 |
    > 155 |   it('should allow example with GitHub URL', async () => {
          |                         ^
      156 |     await usingTempDir(async (cwd) => {
      157 |       const projectName = 'github-app'
      158 |       const res = await run(

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:155:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:153:9)

  ● create next app › should use --example-path over the file path in the GitHub URL

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js github-example-path-2 --example https://github.com/vercel/next-learn-starter/tree/master/navigate-between-pages-starter --example-path navigate-between-pages-starter

      172 |       expect(
      173 |         fs.existsSync(path.join(cwd, projectName, '.gitignore'))
    > 174 |       ).toBeTruthy()
          |                     ^
      175 |       expect(
      176 |         fs.existsSync(path.join(cwd, projectName, 'node_modules/next'))
      177 |       ).toBe(true)

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:174:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:172:9)

  ● create next app › should use npm as the package manager on supplying --use-npm with example

    Command failed with exit code 1 (EPERM): node /home/runner/work/next.js/next.js/packages/create-next-app/dist/index.js use-npm --use-npm --example https://github.com/vercel/next-learn-starter/tree/master/learn-starter

      326 |         '.eslintrc.json',
      327 |       ]
    > 328 |       files.forEach((file) =>
          |                         ^
      329 |         expect(fs.existsSync(path.join(cwd, file))).toBeTruthy()
      330 |       )
      331 |     })

      at makeError (../node_modules/execa/lib/error.js:58:11)
      at handlePromise (../node_modules/execa/index.js:112:26)
      at integration/create-next-app/index.test.js:328:25
      at usingTempDir (integration/create-next-app/index.test.js:23:9)
      at Object.<anonymous> (integration/create-next-app/index.test.js:326:9)

</details>

**Review by timneutkens — APPROVED** [review]

**Pull request merged**